kong/tests/test_unit.py

```diff
@@ -27,11 +27,51 @@
     'kong.stream.status.count': 'monotonic_count',
 }
 
+EXPECTED_METRICS_v3 = {
+    'kong.bandwidth.bytes.count': 'monotonic_count',
+    'kong.http.requests.count': 'monotonic_count',
+    'kong.kong.latency.ms.bucket': 'monotonic_count',
+    'kong.kong.latency.ms.count': 'monotonic_count',
+    'kong.kong.latency.ms.sum': 'monotonic_count',
+    'kong.memory.lua.shared_dict.bytes': 'gauge',
+    'kong.memory.lua.shared_dict.total_bytes': 'gauge',
+    'kong.memory.workers.lua.vms.bytes': 'gauge',
+    'kong.nginx.connections.total': 'gauge',
+    'kong.nginx.requests.total': 'gauge',
+    'kong.nginx.timers': 'gauge',
+    'kong.request.latency.ms.bucket': 'monotonic_count',
+    'kong.request.latency.ms.count': 'monotonic_count',
+    'kong.request.latency.ms.sum': 'monotonic_count',
+    'kong.upstream.latency.ms.bucket': 'monotonic_count',
+    'kong.upstream.latency.ms.count': 'monotonic_count',
+    'kong.upstream.latency.ms.sum': 'monotonic_count',
+}
+
 
 def get_fixture_path(filename):
     return os.path.join(HERE, 'fixtures', filename)
 
 
+def test_check_v3(aggregator, dd_run_check, mock_http_response):
+    mock_http_response(file_path=get_fixture_path('prometheus-v3.txt'))
+    instance = {
+        'openmetrics_endpoint': METRICS_URL,
+        'extra_metrics': [{'kong_memory_workers_lua_vms_bytes': 'memory.workers.lua.vms.bytes'}],
+    }
+
+    check = Kong('kong', {}, [instance])
+    dd_run_check(check)
+
+    for metric_name, metric_type in EXPECTED_METRICS_v3.items():
+        aggregator.assert_metric(metric_name, metric_type=getattr(aggregator, metric_type.upper()))
+
+    aggregator.assert_all_metrics_covered()
+
```

Comment:
Can we try adding this here?

```suggestion
    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
```

(requires `from datadog_checks.dev.utils import get_metadata_metrics`)

That should help us spot errors in the metadata.csv file.

Reply:
Done, still passes!